bsip-0047.md

```diff
@@ -0,0 +1,230 @@
+
+
+### `account_update_votes_operation` (New)
+
+The following is an example implementation.
+
+```
+struct account_update_votes_operation : public base_operation
+{
+	asset fee;
+
+	/// The account to update
+	account_id_type account;
+
+	/// New account options
+	flat_set<vote_id_type>    votes_to_add;
+	flat_set<vote_id_type>    votes_to_remove;
+
+	// A new voting account to set
+	optional<account_id_type> committee_voting_account
+	optional<account_id_type> witness_voting_account
+	optional<account_id_type> worker_voting_account
+
+	// A new number of witness_votes to set
+	optional<uint16_t> num_witness;
+
+	// A new number of committee_votes to set
+	optional<uint16_t> num_committee;
+
+	// For future extensions (see account_update_operation)
+	extensions_type extensions;
+};
+```
+
+Changes to direct votes can be specified by means of a *list* of vote identifiers to be added or removed. This has multiple advantages:
+
+* Reduces the size of the operation compared to replacing the entire slate
+* Allows to easily compare votes before and after the operation
+
+**Validation checks:**
+
+Many of the validation checks are dependent on the merger of the _proposed_ values in the operation with the account's _current_ voting values on the blockchain.
+
+_General validation checks_
+* `account` must be a valid account ID, must exist, must authorize the operation, must pay fee
+* `committee_voting_account`, if present, must be a valid account ID, must exist
+* `witness_voting_account`, if present, must be a valid account ID, must exist
+* `worker_voting_account`, if present, must be a valid account ID, must exist
+* Validate the existence of each vote to add and remove
+
+_Committee validation checks_
+* Let the new _committee_ slate consist of the current _commitee_ slate plus the _committee_ `votes_to_add` minus the _committee_ `votes_to_remove`
+* If `account_options.committee_voting_account == GRAPHENE_PROXY_TO_SELF_ACCOUNT`
+	* Let `new_committee_account` = `GRAPHENE_PROXY_TO_SELF_ACCOUNT`
+* otherwise
+	* Let `new_committee_account` = `account_options.committee_account`
+* If `num_committee` is present
+	* Let `new_num_committee` = `num_committee`
+* otherwise
+	* Let `new_num_committee` = `account_options.num_committee`
+* `num_committee`, if present, is valid:
+	* if `new_committee_account` = `GRAPHENE_PROXY_TO_SELF_ACCOUNT`, and
+	* if 0 &le; `num_committee` &le; `new_num_committee`
+* The _committee_ `votes_to_add` and `votes_to_remove` are valid:
+	* if `new_committee_account` = `GRAPHENE_PROXY_TO_SELF_ACCOUNT`, and
+	* if 0 &le; `new_num_committee` &le; size of new _committee_ slate &le; `chain_params.maximum_committee_count`
+
+_Witness Validation Checks_
+* Let the new _witness_ slate consist of the current _witness_ slate plus the _witness_ `votes_to_add` minus the _witness_ `votes_to_remove`
+* If `account_options.witness_voting_account == GRAPHENE_PROXY_TO_SELF_ACCOUNT`
```

Comment:
I think what you mean is
* If witness_voting_account is present, let new_witness_account = witness_voting_account
* otherwise let new_witness_account = account_options.witness_voting_account